src/Data/Text.hs

```diff
@@ -362,6 +362,16 @@ instance Read Text where
 instance Semigroup Text where
     (<>) = append
 
+    -- | Beware: this function will evaluate to error if the given number does
+    -- not fit into an @Int@.
+    stimes howManyTimes =
+        let howManyTimesInt = P.fromIntegral howManyTimes :: Int
+        in  if P.fromIntegral howManyTimesInt == howManyTimes
+            then replicate howManyTimesInt
+            else P.error "`Data.Text.stimes`: given number does not fit into an `Int`!"
```

Comment:
I wonder if we want to have an exceptional case for an empty `Text`, allowing it to be repeated googolplex times, but I don't have a strong opinion. @Lysxia what do you think?

Reply:
I think that's ok to do but I don't feel strongly about it either.

```suggestion
            else P.error "Data.Text.stimes: given number does not fit into an Int!"
```

to stick with the existing formatting of other error messages in text and base.

Reply:
Right now `stimes` is defined to behave as `replicate` on every input in the domain of `replicate` and undefined elsewhere. This is easy to understand as a principle — a valuable quality. Since neither of you have a strong opinion, I am going to decide and I say we keep it undefined at `""` for now. This is a future-proof decision — we can define it at `""` later if we change our minds.

> to stick with the existing formatting of other error messages in text and base.

Alright, I shall fix this up.

Reply:
Change suggested by @Lysxia is fixed up into the commit history.